### PR TITLE
859 route path node saving

### LINKED ISSUE #859: RoutePath node modification & saving
BLOCKED BY #793 
Continues where PR #793 left, add ability to modify routePath's node. Remember that some last node's information should be saved to routePath instead of routePathLink.

Implement 2 hardcoded values with a checkbox:
[]  Pysäkki ei käytössä, relpysäkki = 'P'
[X] Pysäkki ei käytössä, relpysäkki = 'E'

Blocked until everything is crystal clear in our excel sheet

### PR DESCRIPTION
**Pull request by Divinare**
Closes #859 

BE PR: https://gitlab.hsl.fi/jore/jore-map-backend/merge_requests/62

changes:
- checkbox now sends boolean value at onClick
- removed multiTabArea component (it was legacy code, not needed anymore)
- renamed startNodeType -> startNodeStopType
- ajantasauspysäkki
   * this is a dropdown (Ajantasaus pysakki) with 3 options
- pysäkki ei käytössä
- hastus paikka checkbox
- Kutsu-/jättö/ottopysäkki dropdown
- routePath's bookSchedule information is reset when last link / node is added. Undo / redo doesn't reset that however
- disabled native ctrl + z & ctrl + y because they undoed input fields that caused unwanted behavior

Note: I didn't add "Modified by, modified on" to crossRoads. That is because one can't edit routePath crossroad data them at the moment (they can add new or remove, but they can't edit their properties)

### CODE REVIEW

File: src/models/validationModels/routePathLinkValidationModel.ts [hunk 1]
```diff
@@ -4,9 +4,10 @@ const routePathLinkValidationModel = {
     routePathDirection: 'required|min:1|max:1|string',
     routePathStartDate: 'required|date',
     orderNumber: `required|min:1|max:${Constants.SMALL_INT_MAX_VALUE}|numeric`,
-    startNodeType: 'required|min:1|max:1|string',
     transitType: 'required|min:1|max:1|string',
-    isStartNodeTimeAlignmentStop: 'boolean',
+    startNodeStopType: 'min:1|max:1|string',
+    isStartNodeDisabled: 'required|boolean',
+    startNodeTimeAlignmentStop: 'required|string',
```
Comment (kronken) [inline src/models/validationModels/routePathLinkValidationModel.ts]: does this have some min/max length?

File: src/models/IRoutePathLink.ts [hunk 2]
```diff
@@ -11,13 +10,14 @@ export default interface IRoutePathLink extends IRoutePathLinkPrimaryKey {
...
+    orderNumber: number;
+    transitType: TransitType;
     startNode: INode;
     endNode: INode;
-    orderNumber: number;
-    startNodeType: NodeType;
-    isStartNodeTimeAlignmentStop: boolean;
+    startNodeStopType: string;
+    isStartNodeDisabled: boolean;
+    startNodeTimeAlignmentStop: string; // '0' = false, '1/2' = true
```
[the remark this answers is not in the document]
Comment (Divinare) [reply src/models/IRoutePathLink.ts]: Removed that comment

File: src/models/validationModels/routePathLinkValidationModel.ts (hunk 1 above)
Comment (Divinare) [reply src/models/validationModels/routePathLinkValidationModel.ts]: min:1|max:1

**Comment by Divinare** [comment]
Review fixes done

* removed undoStore clearing. It might not be needed when we have bookScheduleProperties at undoStore.

File: src/components/sidebar/routePathView/routePathListTab/RoutePathListNode.tsx [hunk 3]
```diff
@@ -265,14 +317,12 @@ class RoutePathListNode extends ViewFormBase<
...
-                    <TextContainer
-                        label='MITTAUSPÄIVÄMÄÄRÄ'
-                        value={node.measurementDate}
-                        darkerInputLabel={true}
-                    />
-                </div>
+            <div className={s.flexRow}>
+                <TextContainer
+                    label='MITTAUSPÄIVÄMÄÄRÄ'
+                    value={node.measurementDate}
```
Comment (kronken) [inline src/components/sidebar/routePathView/routePathListTab/RoutePathListNode.tsx]: measurmentDate is in this version string, and that is why this is broken. My PR has it typed to Date, so it should fix the formatting issue

**Pull request merged**